packages/components-kit/src/components/fields/Datalist/Datalist.component.js

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import DatalistWidget from '@talend/react-forms/lib/UIForm/fields/Datalist';
+import { getValue } from '@talend/react-forms/lib/UIForm//utils/properties';
+
+export default function Datalist(props) {
+	function resolveName(value) {
+		// create schema to get entry name from internal properties
+		const key = Array.from(props.schema.key);
+		key[key.length - 1] = `$${key[key.length - 1]}_name`;
+
+		const nameSchema = { ...props.schema, key };
+		return getValue(props.properties, nameSchema) || value;
+	}
+
+	return <DatalistWidget {...props} resolveName={resolveName} />;
```

Comment:
it looks weird to me, props here are available when it called by the Datalist.
So here you create a new function on each render, why not using it this way:
```javascript
function resolveName(value, props) {
...same content
}
function Datalist(props) {
  return <DatalistWidget {...props} resolveName={resolveName} />;
}
```
So you change that it Datalist.component.js

```javascript
.map(value => ({ name: this.props.resolveName(value), value }))
// by
.map(value => ({ name: this.props.resolveName(value, this.props), value }))
```